File: server/paths.ts
import os from "node:os";
import path from "node:path";

export function piAgentHome(): string {
  return path.join(os.homedir(), ".pi", "agent");
}

export function modelsJson(): string {
  return path.join(piAgentHome(), "models.json");
}

export function authJson(): string {
  return path.join(piAgentHome(), "auth.json");
}

export function settingsJson(): string {
  return path.join(piAgentHome(), "settings.json");
}

export function sessionsDir(): string {
  return path.join(piAgentHome(), "sessions");
}

export function skillsDir(): string {
  return path.join(piAgentHome(), "skills");
}

export function agentsSkillsDir(): string {
  return path.join(os.homedir(), ".agents", "skills");
}

export function npmDir(): string {
  return path.join(piAgentHome(), "npm");
}

export function usageLogPath(): string {
  return path.join(piAgentHome(), "pi-switch", "usage-events.jsonl");
}

File: src-tauri/src/paths.rs
use std::path::PathBuf;

/// Resolve the pi agent home directory: `~/.pi/agent`
pub fn pi_agent_home() -> Result<PathBuf, String> {
    let home = dirs::home_dir().ok_or_else(|| "cannot resolve home directory".to_string())?;
    Ok(home.join(".pi").join("agent"))
}

pub fn models_json() -> Result<PathBuf, String> {
    Ok(pi_agent_home()?.join("models.json"))
}

pub fn auth_json() -> Result<PathBuf, String> {
    Ok(pi_agent_home()?.join("auth.json"))
}

pub fn settings_json() -> Result<PathBuf, String> {
    Ok(pi_agent_home()?.join("settings.json"))
}

pub fn sessions_dir() -> Result<PathBuf, String> {
    Ok(pi_agent_home()?.join("sessions"))
}

pub fn skills_dir() -> Result<PathBuf, String> {
    Ok(pi_agent_home()?.join("skills"))
}

pub fn agents_skills_dir() -> Result<PathBuf, String> {
    let home = dirs::home_dir().ok_or_else(|| "cannot resolve home directory".to_string())?;
    Ok(home.join(".agents").join("skills"))
}

pub fn npm_dir() -> Result<PathBuf, String> {
    Ok(pi_agent_home()?.join("npm"))
}

pub fn usage_log_path() -> Result<PathBuf, String> {
    Ok(pi_agent_home()?.join("pi-switch").join("usage-events.jsonl"))
}
